bgpd/bgp_attr_evpn.c

```diff
@@ -0,0 +1,321 @@
+Software Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
+02111-1307, USA.  */
+
+#include <zebra.h>
+
+#include "command.h"
+#include "filter.h"
+#include "prefix.h"
+#include "log.h"
+#include "memory.h"
+#include "stream.h"
+
+#include "bgpd/bgpd.h"
+#include "bgpd/bgp_attr.h"
+#include "bgpd/bgp_route.h"
+#include "bgpd/bgp_attr_evpn.h"
+#include "bgpd/bgp_ecommunity.h"
+#include "bgpd/bgp_evpn.h"
+
+void bgp_add_routermac_ecom (struct attr* attr, char * routermac)
+{
+  struct ecommunity_val routermac_ecom;
+
+  if(attr->extra)
+    {
+      memset(&routermac_ecom, 0, sizeof(struct ecommunity_val));
+      routermac_ecom.val[0] = ECOMMUNITY_ENCODE_EVPN;
+      routermac_ecom.val[1] = ECOMMUNITY_EVPN_SUBTYPE_ROUTERMAC;
+      memcpy(&routermac_ecom.val[2], routermac, MAC_LEN);
+      if(!attr->extra->ecommunity)
+        attr->extra->ecommunity = ecommunity_new ();
+      ecommunity_add_val(attr->extra->ecommunity, &routermac_ecom);
+    }
+}
+
+static uint8_t convertchartohexa (uint8_t *hexa, int *error)
+{
+  if( (*hexa == '0') || (*hexa == '1') || (*hexa == '2') ||
+      (*hexa == '3') || (*hexa == '4') || (*hexa == '5') ||
+      (*hexa == '6') || (*hexa == '7') || (*hexa == '8') ||
+      (*hexa == '9'))
+    return (uint8_t)(*hexa)-'0';
+  if((*hexa == 'a') || (*hexa == 'A'))
+    return 0xa;
+  if((*hexa == 'b') || (*hexa == 'B'))
+    return 0xb;
+  if((*hexa == 'c') || (*hexa == 'C'))
+    return 0xc;
+  if((*hexa == 'd') || (*hexa == 'D'))
+    return 0xd;
+  if((*hexa == 'e') || (*hexa == 'E'))
+    return 0xe;
+  if((*hexa == 'f') || (*hexa == 'F'))
+    return 0xf;
+  *error = -1;
+  return 0;
+}
+
+/* converts to internal representation of mac address
+ * returns 1 on success, 0 otherwise 
+ * format accepted: AA:BB:CC:DD:EE:FF
+ * if mac parameter is null, then check only
+ */
+int
+str2mac (const char *str, char *mac)
+{
+  unsigned int k=0, i, j;
+  uint8_t *ptr, *ptr2;
+  size_t len;
+  uint8_t car;
+
+  if (!str)
+    return 0;
+
+  if (str[0] == ':' && str[1] == '\0')
+    return 1;
+
+  i = 0;
+  ptr = (uint8_t *)str;
+  while (i < 6)
+    {
+      uint8_t temp[5];
+      int error = 0;
+      ptr2 = (uint8_t *)strchr((const char *)ptr, ':');
+      if (ptr2 == NULL)
+	{
+	  /* if last occurence return ok */
+	  if(i != 5)
+            {
+              zlog_err("[%s]: format non recognized",mac);
+              return 0;
+            }
+          len = strlen((char *)ptr);
+	} 
+      else
+        {
+          len = ptr2 - ptr;
+        }
+      if(len > 5)
+        {
+          zlog_err("[%s]: format non recognized",mac);
+         return 0;
+        }
+      memcpy(temp, ptr, len);
+      for(j=0;j< len;j++)
+	{
+	  if (k >= MAC_LEN)
+	    return 0;
+          if(mac)
+            mac[k] = 0;
+          car = convertchartohexa (&temp[j], &error);
+	  if (error)
+	    return 0;
+	  if(mac)
+            mac[k] = car << 4;
+	  j++;
+          if(j == len)
```

Comment:
hello @qlyoung , 
I assume the indentation is general to the whole new files added.
I will add a commit taking into account the result of indent following options:

-nbad -bap -nbc -bbo -hnl -br -brs -c33 -cd33 -ncdb -ce -ci4
-cli0 -d0 -di1 -nfc1 -i8 -ip0 -l80 -lp -npcs -nprs -npsl -sai
-saf -saw -ncs -nsc -sob -nfca -cp33 -ss -ts8 -il1

please tell me when the commit is present, if this is what you expect ?